c++/src/kj/async-io-test.c++

```diff
@@ -1634,6 +1634,27 @@ KJ_TEST("Userland pipe tryPumpFrom to pumpTo for same amount fulfills simultaneo
   KJ_EXPECT(pumpPromise2.wait(ws) == 6);
 }
 
+KJ_TEST("Userland pipe multi-part write doesn't quit early") {
+  kj::EventLoop loop;
+  WaitScope ws(loop);
+
+  auto pipe = newOneWayPipe();
+  auto pipe2 = newOneWayPipe();
+  auto pumpPromise = KJ_ASSERT_NONNULL(pipe2.out->tryPumpFrom(*pipe.in, 6));
```

Comment:
Heh, indeed, I think it's actually wrong... `pipe.in` ends up with multiple logical readers, but it doesn't end up causing a problem because the `tryPumpFrom()` stays blocked waiting for a reader on `pipe2.in`, which never comes. Whoops.